---
layout: archive
title: Zulip Chat Archive
permalink: /stream/113488-general/topic/parser.2Eof_tactic.20is.20broken.20when.20the.20tactic.20fails.html
---

<h2>Stream: <a href="https://leanprover-community.github.io/archive/stream/113488-general/index.html">general</a></h2>
<h3>Topic: <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/parser.2Eof_tactic.20is.20broken.20when.20the.20tactic.20fails.html">parser.of_tactic is broken when the tactic fails</a></h3>

<hr>

<base href="https://leanprover.zulipchat.com/">

<head><link href="/style.css" rel="stylesheet"></head>

{% raw %}

<a name="133093024"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/parser.of_tactic%20is%20broken%20when%20the%20tactic%20fails/near/133093024" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Mario Carneiro <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/parser.2Eof_tactic.20is.20broken.20when.20the.20tactic.20fails.html#133093024">(Aug 30 2018 at 23:36)</a>:</h4>
<p>I just stumbled on the following issue:</p>
<div class="codehilite"><pre><span></span>open tactic
meta def test : tactic unit := failed
meta def tac (_ : interactive.parse test) := skip
run_cmd add_interactive [`tac]
example : true := by tac.
-- vm check failed: is_closure(o) (possibly due to incorrect axioms, or sorry)
</pre></div>


<p>The problem appears to be that nothing catches an error thrown in a tactic when it is passed through the <code>of_tactic</code> builtin function</p>



{% endraw %}

<hr><p>Last updated: Jan 25 2023 at 00:06 UTC</p>